src/main/java/org/mifos/connector/common/interceptor/JWSUtil.java

```diff
@@ -141,8 +144,19 @@ public static String parseBodyPayload(HttpServletRequest request) throws IOExcep
      * @throws IOException
      */
     public static String parseFormData(HttpServletRequest httpServletRequest) throws ServletException, IOException {
+        log.debug("Parsing form data");
         StringBuilder stringBuilder = new StringBuilder();
-        for (Part part: httpServletRequest.getParts()) {
+        Collection<Part> parts;
+        try {
+            parts = httpServletRequest.getParts();
+            assert parts != null;
```

Comment:
Instead of assert, handle validations in standard way

Reply:
httpServletRequest.getParts() 

is flaky it throws `IOException, ServletException` sometimes and sometimes it works, thats why Its done like this.

Reply:
For more read https://stackoverflow.com/questions/4537609/is-it-good-practice-to-use-assert-in-java#4538090

Reply:
done